locator/everywhere_replication_strategy.cc

```diff
@@ -40,6 +40,15 @@ void everywhere_replication_strategy::validate_options(const gms::feature_servic
     }
 }
 
+sstring everywhere_replication_strategy::sanity_check_read_replicas(const effective_replication_map& erm, const inet_address_vector_replica_set& read_replicas) const {
+    const auto replication_factor = erm.get_replication_factor();
+    if (read_replicas.size() > replication_factor) {
+        return seastar::format("everywhere_replication_strategy: the number of replicas for everywhere_replication_strategy is {}, cannot be higher than replication factor {}", read_replicas.size(), replication_factor);
+    }
+    return {};
```

Comment:
No, should not prevent RVO as all returned are unnamed values. But if I used `optional` then RVO would had been prevented, and the string would had been moved instead, see [godbolt](https://godbolt.org/z/c8E6vxTb9)
But, I agree, 99% of the time, clarity is more important than performance.

Reply:
While `move` itself is not that bad if the object is movable, notice 2 calls to destructors there, it could be a an issue in a hot path as usually destructors do some checks to find if some resources need to be de-allocated and therefore unwanted branches can be introduced.